src/openms/include/OpenMS/ANALYSIS/ID/FalseDiscoveryRate.h

```diff
@@ -114,6 +158,488 @@ namespace OpenMS
     /// calculates the FDR, given two vectors of scores
     void calculateFDRs_(std::map<double, double>& score_to_fdr, std::vector<double>& target_scores, std::vector<double>& decoy_scores, bool q_value, bool higher_score_better) const;
 
+    template<typename T> struct IsIDType
+    {
+      static bool const value = std::is_same<T, PeptideIdentification>::value || std::is_same<T, ProteinIdentification>::value;
+    };
+
+    template<typename T> struct IsHitType
+    {
+      static bool const value = std::is_same<T, PeptideHit>::value || std::is_same<T, ProteinHit>::value;
+    };
+    /*
```

Comment:
what about this bunch of commented out code?

Reply:
even better: do it in the class docu!

Reply:
removed